src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooService.java

```diff
@@ -82,8 +84,22 @@ public BambooService(GitService gitService, ProgrammingSubmissionRepository prog
     @Override
     public void createBuildPlanForExercise(ProgrammingExercise programmingExercise, String planKey, VcsRepositoryUrl sourceCodeRepositoryURL, VcsRepositoryUrl testRepositoryURL,
             VcsRepositoryUrl solutionRepositoryURL) {
+        var additionalRepositories = programmingExercise.getAuxiliaryRepositoriesForBuildPlan()
+            .stream()
+            .map(repo -> Pair.of(repo.getName(), urlService.getRepositorySlugFromRepositoryUrl(repo.getVcsRepositoryUrl())))
```

Comment:
Those `Pair` usages would be a perfect fit for a Java record. Consider switching to that (way more expressive than `Pair<String, String>`).

*There a multiple places here where this is used, I only happened to add the comment here.*